src/main/java/net/fabricmc/loader/impl/gui/FabricMainWindow.java

```diff
@@ -89,11 +92,29 @@ private static void open0(FabricStatusTree tree, boolean shouldWait) throws Exce
 		}
 	}
 
+	private static String getTitleVersion() {
+		Optional<ModContainer> optional = FabricLoader.getInstance().getModContainer("fabricloader");
+		if (optional.isPresent()) {
```

Comment:
If fabric loader cannot find the `fabricloader` mod (also is there a constant for this mod id?), the loader is most likely in an error state and doomed to fail any way. So I suggest an `orElseThrow(() -> new IllegalStateException("cannot find the loader mod"))` etc

Reply:
There is not a constant for the mod ID, but I think that would be worth adding in a later style PR.

Reply:
I think the graceful handling is better, this really shouldn't break..